fdbclient/NativeAPI.actor.cpp

```diff
@@ -3189,6 +3189,22 @@ Future< StorageMetrics > Transaction::getStorageMetrics( KeyRange const& keys, i
 	return ::waitStorageMetrics( cx, keys, StorageMetrics(), m, StorageMetrics(), shardLimit );
 }
 
+ACTOR Future< Standalone<RangeResultRef> > waitStorageMetricsList(
+	Database cx,
+	KeyRange keys,
+	int shardLimit )
+{
+	ErrorOr<GetDDMetricsReply> rep = wait(errorOr(loadBalance(cx->getMasterProxies(false), &MasterProxyInterface::getDDMetrics, GetDDMetricsRequest(keys, shardLimit))));
```

Comment:
If the proxies change, this request should be resent. See `getHealthMetricsActor` for an example of how to deal with this. Also, if this request times out, a timeout should be thrown (i.e. don't wait for a timeout error to be received from the proxy, because a slow proxy might be the reason a request should time out).